decision-records/006-javascript-compatibility.md

```diff
@@ -0,0 +1,161 @@
+# JavaScript browser compatibility
+
+Date: 27/06/23
+
+Status: Accepted
+
+## Decision
+
+We will ship JavaScript that can parse without errors in browsers supporting `<script type="module">`.
+
+We will update our guidance to tell service teams to include the JavaScript from GOV.UK Frontend using `<script type="module">`.
+
+We will set up:
+
+- transpiling to transform modern code down to that baseline
+- linting to prevent us from using newer features and syntaxes
```

Comment:
Do we need to clarify that we _can_ use newer features where transpiling helps us out?

```suggestion
- linting to prevent us using modern JavaScript that cannot be transpiled
```

Reply:
Rephrased to describe that we still intend to opt in to newer features, as it's not so much that they can't be transpiled, more that we decide not to transpile them.